### ISSUE TITLE
[DOC] Default value of vrrp.preempt parameter

### ISSUE THREAD

**Issue by ipspace** [ipspace/netlab#1420]
The discussions in #1418 and #1419 turned out another omission of mine: what is the default value of **vrrp.preempt** parameter.

I usually assume that without a very good reason to be otherwise, a missing parameter means `False` (thus, the default value of a parameter is `False`). Alternatively, we could interpret the lack of **vrrp.preempt** parameter to mean "leave the stuff alone, whatever the vendor default is is OK"

The alternate approach might lead to a lab having different behavior when ran with different devices, so if nobody forcefully disagrees (cc @jbemmel @ssasso), I'll adapt the documentation of the gateway module to say "default: False", and then we have to go through the various configuration templates checking what the vendors think the default value should be.

**Comment by sdargoeuves** [comment]
I like the approach of saying the default is False, let's bring consistency where vendors bring chaos :)

**Comment by ipspace** [comment]
> I like the approach of saying the default is False, let's bring consistency where vendors bring chaos :)

👍

> I will check some templates and update my PR

Thank you! However, let's wait a day or two to see if someone has any good objections to this approach.

**Comment by ipspace** [comment]
> The few vendors that I roughly checked seem to assume preempt=True as default

Listing them would help ;)

> I would suggest to take an inventory, and set the default to whatever the majority do

Yeah, I have no problem with default being "True", we just have to decide what it should be, document that, and adjust the templates.

**Comment by sdargoeuves** [comment]
I will have a look at the vendors where netlab support vrrp, and come back with a list

**Comment by ipspace** [comment]
> I will have a look at the vendors where netlab support vrrp, and come back with a list

Thanks a million!

**Comment by sdargoeuves** [comment]
Here is the result of my investigations:

## Listing of default value

| Platform | Platform default | netlab default |
|:---------|:-----------------|:---------------|
| ArubaCX | Enabled | **disabled** |
| Cumulus | Enabled | code missing to disable -> **enabled** |
| DellOS | Enabled (jbemmel) | **disabled** |
| EOS | Enabled | code missing to disable -> **enabled** |
| FRR | Enabled | code missing to disable -> **enabled** |
| IOS | Enabled | code missing to disable -> **enabled** |
| NXOS | guessing enabled | code missing to disable -> **enabled** |
| SRLinux | N/A | N/A |
| SROS | ? | **disabled** |
| VyOS | Enabled | **disabled** |

Indeed, It looks like the default for most platforms is to have preempt enabled. 
@ipspace should we change the default value for `gateway.vrrp.preempt` to True, as well as adding the commands where missing?
I think I've code most of the code ready, waiting for the confirmation before doing further tests, then I can update the PR.

**Comment by sdargoeuves** [comment]
Interestingly, the code for dellos10, uses the fact that the platform uses preempt as default, to only remove the preempt if the `vrrp.preeempt` is not True.

```j2
{%       if not intf.gateway.vrrp.preempt|default(False) %}
    no preempt
```

Should we force either commands depending on the value of vrrp.preempt (which I think would be better, so we know for sure what will be configured) or do we only remove preempt when it's the default for the platform?

**Comment by ipspace** [comment]
> Here is the result of my investigations:

Awesome job. Thanks a million!!!

> Indeed, It looks like the default for most platforms is to have preempt enabled. @ipspace should we change the default value for `gateway.vrrp.preempt` to True

**Yes.**

> as well as adding the commands where missing?

Yes.

> I think I've code most of the code ready, waiting for the confirmation before doing further tests, then I can update the PR.

Please do. I can add the documentation changes to the PR before merging it.

**Comment by ipspace** [comment]
> Should we force either commands depending on the value of vrrp.preempt (which I think would be better, so we know for sure what will be configured) 

A year ago I would have said "we should just configure the non-default state" to make it idempotent. However, I don't think too many people run **netlab initial** three times in a row or check whether anything has changes, so let's go for "knowing for sure".

**Comment by ipspace** [comment]
Fixed in #1419
